pieces.py

```diff
@@ -299,78 +282,55 @@ def _get_diagonal_valid_moves(piece):
     return valid_moves
 
 
-def _get_horizontal_valid_move_squares(piece):
-    this_tur = piece.square.tur_cord
-
+def _get_straight_valid_move_squares(piece, is_vertical):
+    piece_tur = piece.square.tur_cord
+    piece_line = piece.square.line_cord
+    move_on = piece_tur if is_vertical else piece_line
+    permanent = piece_line if is_vertical else piece_tur
     valid_moves = []
 
-    for line in range(BOARD_LINE):
-        if square_is_valid(this_tur, line, piece.is_in_white_team):
-            square = squares[line][this_tur]
+    for tur_or_line in range(move_on):
+        square = squares[permanent][tur_or_line] if is_vertical else squares[tur_or_line][permanent]
+        if square.current_piece is None:
             # If square is empty we just add him to the row.
             valid_moves.append(square)
-            if square.current_piece is not None:
-                # If square is taken by enemy and including piece(this rook), this is the valid squares.
-                if piece.square in valid_moves:
-                    break
-                # Other we start a new raw from this square(including).
-                else:
-                    valid_moves = [square]
             continue
-
-        # Because piece.team equals to piece.team he isn't valid but we need to add him to return square.
-        if squares[line][this_tur].current_piece == piece:
-            valid_moves.append(piece.square)
-
-        # If the row of squares touch teammate square and including piece(this piece), this is the valid squares.
-        elif piece.square in valid_moves:
-            break
-
-        # Other we have to make a new row.
-        else:
+        # If square taken by teamate piece we have to make a new row.
+        if square.current_piece.team is piece.team:
             valid_moves = []
+        else:
+            # square is taken by enemy.
+            # we start a new raw from this square(including).
+            valid_moves = [square]
+            continue
 
-    valid_moves.remove(piece.square)
-    return valid_moves
-
-
-def _get_vertical_valid_move_squares(piece):
-    # Check squares left  and right then root.
-
-    this_line = piece.square.line_cord
-    valid_moves = []
-
-    for tur in range(BOARD_LINE):
-        if square_is_valid(tur, this_line, piece.is_in_white_team):
-            square = squares[this_line][tur]
+        # Add squares after piece
+    for tur_or_line in range(move_on+1, BOARD_LINE):
+        square = squares[permanent][tur_or_line] if is_vertical else squares[tur_or_line][permanent]
+        if square.current_piece is None:
+            # If square is empty we just add him to the row.
             valid_moves.append(square)
-            if square.current_piece is not None:
-                if piece.square in valid_moves:
-                    break
-                else:
-                    valid_moves = [square]
             continue
+        # square is taken by enemy.
```

Comment:
Delete comment, you have it in line 333.

Reply:
:+1: